packages/rum/src/boot/recorderApi.ts

```diff
@@ -63,6 +63,11 @@ export function makeRecorderApi(startRecordingImpl: StartRecording): RecorderApi
         if (state.status === RecorderStatus.IntentToStart) {
           startStrategy()
         }
+
+        if (!session.hasReplayPlan() && state.status === RecorderStatus.Started) {
+          stopStrategy()
+          state = { status: RecorderStatus.IntentToStart }
+        }
```

Comment:
I am wondering if it would make sense to always re-call stop/start at session renew.
Could the triggered lifecycle events be needed?

Reply:
From what I see,  RECORD_STARTED and RECORD_STOPPED are only subscribed in view tracking to update the hasReplay flag. If the plan doesn't change between two session there no need to trigger them. 
In addition, it does not really simplify the code. 
Maybe it could make sense in term of domain logic but for now, I don't see it.

Reply:
> RECORD_STARTED and RECORD_STOPPED are only subscribed in view tracking to update the hasReplay flag. 

yes and since this flag is not reset on renew session it works fine

>  it does not really simplify the code

it could remove the need to check the plan so one less concept to have in mind to understand what is happening


Also, could we end up with a segment mixing events from different sessions if we don't flush the current one on session renew?

Reply:
> it could remove the need to check the plan so one less concept to have in mind to understand what is happening

Yes, we might end up with:       
```js
lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
  const status = state.status
  stopStrategy()

  if (status !== RecorderStatus.Stopped) {
    startStrategy()
  } 
})
```
> Also, could we end up with a segment mixing events from different sessions if we don't flush the current one on session renew?

FMU, the recorder only cares about view life cycle. SESSION_RENEWED event triggers VIEW_ENDED and VIEW_CREATED, which will be caught by session replay to flushSegments. 
https://github.com/DataDog/browser-sdk/blob/80d25788d40ccaffcb8377927e8bdd4a88b4f14a/packages/rum/src/domain/segmentCollection/segmentCollection.ts#L95-L97
New segments are flagged with the new session id found in the same parentContexts has RUM as you can see here.
https://github.com/DataDog/browser-sdk/blob/80d25788d40ccaffcb8377927e8bdd4a88b4f14a/packages/rum/src/domain/segmentCollection/segmentCollection.ts#L191-L200

The parent context is up to date because it also listen VIEW_CREATED
https://github.com/DataDog/browser-sdk/blob/80d25788d40ccaffcb8377927e8bdd4a88b4f14a/packages/rum-core/src/domain/parentContexts.ts#L39-L42

Reply:
ok